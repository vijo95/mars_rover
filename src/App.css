/* RoverCar */
.rover-card{
  cursor: pointer;
  margin: 20px !important;
  transition: 0.2s ease-in-out !important;
}

.rover-card:hover{
  transform: scale(1.1);
  opacity: 0.8;
}

.rover-selected{
  border: 5px solid #00BFA5 !important;
}


/* ListPhotos */
.grid-container{
  column-count: 3;
}

.img_wrap_grid {
  position: relative;
  border-radius: 5px;
  text-align: center;
  margin: auto;
  margin-bottom: 20px;
  justify-content: space-evenly;
}

.img-post{
  object-fit: cover;
  border-radius: 5px;
}

@media (max-width:550px){
  .grid-container{
    column-count: 2
  }
}

@media (max-width:319px){
  .grid-container{
    column-count: 1
  }
}