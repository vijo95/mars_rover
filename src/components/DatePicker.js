import React, { Component } from 'react'
import DayPickerInput from 'react-day-picker/DayPickerInput';
import 'react-day-picker/lib/style.css';


export default class DatePicker extends Component {


  parseDate = (day) => {
    const date = day.toLocaleDateString().split("/")
    this.props.setDateType('earth_date')
    this.props.setEarthDate(`${date[2]}-${date[0]}-${date[1]}`)
    this.props.setSolDate('')
    return `${date[2]}-${date[0]}-${date[1]}`
  }

  constructor(props) {
    super(props);
    this.handleDayChange = this.handleDayChange.bind(this);
    this.state = {
      selectedDay: null,
    };
  }

  handleDayChange(day) {
    try {
      this.setState({ selectedDay: this.parseDate(day)});
    } catch (error) {
      console.log(error);
    }
  }

  render() {

    return (
      <DayPickerInput 
        value={this.props.earthDate}
        placeholder="Earth date"
        onDayChange={this.handleDayChange} />
    )
  }
}
