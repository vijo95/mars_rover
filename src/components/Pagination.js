import React from 'react'
import { Icon, Label } from 'semantic-ui-react'

export default function Pagination({page, setPage, rover, fetchPhotos}) {

  function handlePageChange(n) {
    if(1 <= page+n){
      setPage(page+n)
      fetchPhotos(rover, page+n)
    }
  }



  return (
    <div style={{textAlign:'center', margin:'20px'}}>
      <Icon 
        link name="angle double left" 
        size="big"
        onClick={() => handlePageChange(-10)}
      />

      <Icon 
        link name="angle left" 
        size="big"
        onClick={() => handlePageChange(-1)}
      />

      <Label circular size="huge">
        {page}
      </Label>

      <Icon 
        link name="angle right"  
        size="big"
        onClick={() => handlePageChange(1)}
      />

      <Icon 
        link name="angle double right"  
        size="big"
        onClick={() => handlePageChange(10)}
      />
    </div>
  )
}
