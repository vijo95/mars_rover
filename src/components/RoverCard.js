import React from 'react'
import { Card, Image } from 'semantic-ui-react'

export default function RoverCard({img,name,fetchPhotos,setRover, rover, page}) {

  function handleRoverChange(){
    setRover(name)
    fetchPhotos(name, page)
  }

  return (
    <Card className={rover === name ? "rover-card rover-selected" : "rover-card"} 
      onClick={(e) => handleRoverChange()}>
      <Image src={img} wrapped ui={true} />
      <Card.Content>
        <Card.Header textAlign="center">
          {name ? name.charAt(0).toUpperCase() + name.slice(1) : ''}
        </Card.Header>
      </Card.Content>
    </Card>
  )
}
