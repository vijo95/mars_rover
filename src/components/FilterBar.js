import React from 'react'
import { Button, Dropdown, Menu, Input } from 'semantic-ui-react'
import DatePicker from './DatePicker'

export default function FilterBar({setDateType,setSolDate,setEarthDate,setCamera,
  solDate,earthDate,camera,fetchPhotos,rover}) {

    function handleSolChange(event) {
      event.preventDefault()
      setSolDate(parseInt(event.target.value))
      setDateType("sol")
      setEarthDate('')
      setCamera('')
    }

    function handleCameraChange(name){
      if(name !== ''){
        setCamera(`&camera=${name}`)
      }else{
        setCamera(``)
      }
    }

    return (
      <Menu size='large' style={{maxWidth:'800px', margin:'20px auto'}}>

        <Menu.Item>
          <Input 
            value={solDate}
            onChange={(event) => handleSolChange(event)}
            placeholder='Sol date' 
          />
        </Menu.Item>

        <Menu.Item>
          <DatePicker 
            setDateType={setDateType}
            setEarthDate={setEarthDate}
            setSolDate={setSolDate}
            earthDate={earthDate}
          />
        </Menu.Item>

        <Menu.Menu>
          <Dropdown item text={camera ? camera.split("=")[1].toUpperCase() : "Camera"}>
            <Dropdown.Menu>
              <Dropdown.Item onClick={() => handleCameraChange('')}>Default</Dropdown.Item>
              <Dropdown.Item onClick={() => handleCameraChange('fhaz')}>Front Hazard Avoidance Camera</Dropdown.Item>
              <Dropdown.Item onClick={() => handleCameraChange('rhaz')}>Rear Hazard Avoidance Camera</Dropdown.Item>
              <Dropdown.Item onClick={() => handleCameraChange('mast')}>Mast Camera</Dropdown.Item>
              <Dropdown.Item onClick={() => handleCameraChange('chemcam')}>Chemistry and Camera Complex</Dropdown.Item>
              <Dropdown.Item onClick={() => handleCameraChange('mahli')}>Mars Hand Lens Imager</Dropdown.Item>
              <Dropdown.Item onClick={() => handleCameraChange('mardi')}>Mars Descent Imager</Dropdown.Item>
              <Dropdown.Item onClick={() => handleCameraChange('navcam')}>Navigation Camera</Dropdown.Item>
              <Dropdown.Item onClick={() => handleCameraChange('pancam')}>Panoramic Camera</Dropdown.Item>
              <Dropdown.Item onClick={() => handleCameraChange('minites')}>Miniature Thermal Emission Spectrometer (Mini-TES)</Dropdown.Item>
            </Dropdown.Menu>
          </Dropdown>
        </Menu.Menu>

        <Menu.Item position="right">
          <Button primary
            onClick={() => fetchPhotos(rover)}
          >
            Filtrar
          </Button>
        </Menu.Item>

      </Menu>
    )
}