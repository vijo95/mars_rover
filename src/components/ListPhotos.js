import React from 'react'
import Pagination from './Pagination'
import SinglePhoto from './SinglePhoto'

export default function ListPhotos({photos, page, setPage, rover,fetchPhotos}) {
  return (
    <>
      <Pagination 
        page={page}
        setPage={setPage}
        rover={rover}
        fetchPhotos={fetchPhotos}
      />
      <div style={{margin:'auto',padding:'5px', maxWidth:'800px'}}>
        <div className="grid-container">
          { photos ? 
            photos.length === 0 ?
            <h1>Nothing to see here..</h1> :
            photos.map((photo,index) =>
            <SinglePhoto key={index}
              img={photo.img_src}
            />) : null
          }
        </div>
      </div>
    </>
  )
}
