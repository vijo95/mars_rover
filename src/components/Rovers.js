import React from 'react'
import { Card } from 'semantic-ui-react'
import RoverCard from './RoverCard'

export default function Rovers({fetchPhotos,setRover,rover, page}) {
  return (
    <Card.Group centered>
      <RoverCard
        img={"https://mars.nasa.gov/system/feature_items/images/6037_msl_banner.jpg"}
        name={"curiosity"}
        fetchPhotos={fetchPhotos}
        setRover={setRover}
        rover={rover}
        page={page}
      />
      <RoverCard 
        img={"https://s1.eestatic.com/2019/02/13/actualidad/Actualidad_375974696_114650383_1024x576.jpg"}
        name={"opportunity"}
        fetchPhotos={fetchPhotos}
        setRover={setRover}
        rover={rover}
        page={page}
      />
      <RoverCard 
        img={"https://jplopez.net/wp-content/uploads/2019/09/rover-spirit.jpg"}
        name={"spirit"}
        fetchPhotos={fetchPhotos}
        setRover={setRover}
        rover={rover}
        page={page}
      />
    </Card.Group>
  )
}
