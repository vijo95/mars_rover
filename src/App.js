import React, {useState} from 'react'
import axios from 'axios'
import { Loader } from 'semantic-ui-react'

import Rovers from './components/Rovers'
import FilterBar from './components/FilterBar'
import ListPhotos from './components/ListPhotos';
import { baseAPI, API_KEY } from './utils'
import './App.css';

function App() {

  const [photos, setPhotos] = useState(null)
  const [page, setPage] = useState(1)
  const [solDate, setSolDate] = useState(2000)
  const [earthDate, setEarthDate] = useState('')
  const [dateType, setDateType] = useState("sol")
  const [camera, setCamera] = useState('')
  const [rover, setRover] = useState('curiosity')
  const [loading, setLoading] = useState(false)

  function fetchPhotos(roverName, pageNumber) {
    setLoading(true)
    axios
    .get(baseAPI + roverName + `/photos?${dateType}=${solDate}${earthDate}${camera}&page=${pageNumber}&` + API_KEY)
    .then(res => {
      setPhotos(res.data.photos)
      setLoading(false)
    })
    .catch(err => {
      console.log(err);
    })
  }


  return (
    <>
      <h1 style={{margin:'20px', textAlign:'center', color:'whitesmoke'}}>
        Mars Rover Photos
      </h1>
      
      <Rovers 
        fetchPhotos={fetchPhotos}
        setRover={setRover}
        rover={rover}
        page={page}
      />

      <FilterBar 
        setDateType={setDateType}
        setSolDate={setSolDate}
        setEarthDate={setEarthDate}
        setCamera={setCamera}

        solDate={solDate}
        earthDate={earthDate}
        camera={camera}

        fetchPhotos={fetchPhotos}
        rover={rover}
        page={page}
      />

      { !loading && photos ?
        <ListPhotos 
          photos={photos}
          setPage={setPage}

          fetchPhotos={fetchPhotos}
          rover={rover}
          page={page}
        /> : 
        loading ? <Loader active inline='centered' /> :
        null
      }
    </>
  );
}

export default App;
