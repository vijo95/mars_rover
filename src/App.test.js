import { render } from '@testing-library/react';
import renderer from 'react-test-renderer'
import App from './App';
import SinglePhoto from '../src/components/SinglePhoto'
import Rovers from '../src/components/Rovers'
import RoverCard from '../src/components/RoverCard'
import Pagination from '../src/components/Pagination'
import ListPhotos from '../src/components/ListPhotos'
import FilterBar from '../src/components/FilterBar'
import DatePicker from './components/DatePicker';

// Render Test
test('renders App', () => {
  render(<App />)
});

test('renders SinglePhoto', () => {
  render(<SinglePhoto />)
})

test('renders Rovers', () => {
  render(<Rovers />)
})

test('renders RoverCard', () => {
  render(<RoverCard />)
})

test('renders Pagination', () => {
  render(<Pagination />)
})

test('renders ListPhotos', () => {
  render(<ListPhotos />)
})

test('renders FilterBar', () => {
  render(<FilterBar />)
})

test('renders DatePicker', () => {
  render(<DatePicker />)
})

//Snapshot Test
it("matches snapshot App", () => {
  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
})

it("matches snapshot SinglePhoto", () => {
  const tree = renderer.create(<SinglePhoto />).toJSON()
  expect(tree).toMatchSnapshot()
})

it("matches snapshot Rovers", () => {
  const tree = renderer.create(<Rovers />).toJSON()
  expect(tree).toMatchSnapshot()
})

it("matches snapshot RoverCard", () => {
  const tree = renderer.create(<RoverCard />).toJSON()
  expect(tree).toMatchSnapshot()
})

it("matches snapshot Pagination", () => {
  const tree = renderer.create(<Pagination />).toJSON()
  expect(tree).toMatchSnapshot()
})

it("matches snapshot ListPhotos", () => {
  const tree = renderer.create(<ListPhotos />).toJSON()
  expect(tree).toMatchSnapshot()
})

it("matches snapshot FilterBar", () => {
  const tree = renderer.create(<FilterBar />).toJSON()
  expect(tree).toMatchSnapshot()
})

it("matches snapshot DatePicker", () => {
  const tree = renderer.create(<DatePicker />).toJSON()
  expect(tree).toMatchSnapshot()
})