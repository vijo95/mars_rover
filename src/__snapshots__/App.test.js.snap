// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot App 1`] = `
Array [
  <h1
    style={
      Object {
        "color": "#263238",
        "margin": "20px",
        "textAlign": "center",
      }
    }
  >
    Mars Rover Photos
  </h1>,
  <div
    className="ui centered cards"
  >
    <a
      className="ui card rover-card rover-selected"
      onClick={[Function]}
    >
      <div
        className="ui image"
      >
        <img
          src="https://mars.nasa.gov/system/feature_items/images/6037_msl_banner.jpg"
        />
      </div>
      <div
        className="content"
      >
        <div
          className="center aligned header"
        >
          Curiosity
        </div>
      </div>
    </a>
    <a
      className="ui card rover-card"
      onClick={[Function]}
    >
      <div
        className="ui image"
      >
        <img
          src="https://s1.eestatic.com/2019/02/13/actualidad/Actualidad_375974696_114650383_1024x576.jpg"
        />
      </div>
      <div
        className="content"
      >
        <div
          className="center aligned header"
        >
          Opportunity
        </div>
      </div>
    </a>
    <a
      className="ui card rover-card"
      onClick={[Function]}
    >
      <div
        className="ui image"
      >
        <img
          src="https://jplopez.net/wp-content/uploads/2019/09/rover-spirit.jpg"
        />
      </div>
      <div
        className="content"
      >
        <div
          className="center aligned header"
        >
          Spirit
        </div>
      </div>
    </a>
  </div>,
  <div
    className="ui large secondary stackable menu"
    style={
      Object {
        "margin": "40px auto",
        "maxWidth": "800px",
      }
    }
  >
    <div
      className="item"
      onClick={[Function]}
    >
      <div
        className="ui input"
      >
        <input
          onChange={[Function]}
          placeholder="Sol date"
          type="text"
          value={2000}
        />
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      <div
        className="DayPickerInput"
      >
        <input
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          placeholder="YYYY-MM-DD"
          value=""
        />
      </div>
    </div>
    <div
      className="menu"
    >
      <div
        aria-expanded={false}
        className="ui item top left pointing dropdown"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="listbox"
        tabIndex={0}
      >
        <div
          aria-atomic={true}
          aria-live="polite"
          className="divider text"
          role="alert"
        >
          Camera
        </div>
        <i
          aria-hidden="true"
          className="dropdown icon"
          onClick={[Function]}
        />
        <div
          className="menu transition"
        >
          <div
            className="item"
            onClick={[Function]}
            role="option"
          >
            Default
          </div>
          <div
            className="item"
            onClick={[Function]}
            role="option"
          >
            Front Hazard Avoidance Camera
          </div>
          <div
            className="item"
            onClick={[Function]}
            role="option"
          >
            Rear Hazard Avoidance Camera
          </div>
          <div
            className="item"
            onClick={[Function]}
            role="option"
          >
            Mast Camera
          </div>
          <div
            className="item"
            onClick={[Function]}
            role="option"
          >
            Chemistry and Camera Complex
          </div>
          <div
            className="item"
            onClick={[Function]}
            role="option"
          >
            Mars Hand Lens Imager
          </div>
          <div
            className="item"
            onClick={[Function]}
            role="option"
          >
            Mars Descent Imager
          </div>
          <div
            className="item"
            onClick={[Function]}
            role="option"
          >
            Navigation Camera
          </div>
          <div
            className="item"
            onClick={[Function]}
            role="option"
          >
            Panoramic Camera
          </div>
          <div
            className="item"
            onClick={[Function]}
            role="option"
          >
            Miniature Thermal Emission Spectrometer (Mini-TES)
          </div>
        </div>
      </div>
    </div>
    <div
      className="right item"
      onClick={[Function]}
    >
      <button
        className="ui primary button"
        onClick={[Function]}
      >
         Filtrar
      </button>
    </div>
  </div>,
]
`;

exports[`matches snapshot DatePicker 1`] = `
<div
  className="DayPickerInput"
>
  <input
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    placeholder="YYYY-MM-DD"
    value=""
  />
</div>
`;

exports[`matches snapshot FilterBar 1`] = `
<div
  className="ui large secondary stackable menu"
  style={
    Object {
      "margin": "40px auto",
      "maxWidth": "800px",
    }
  }
>
  <div
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui input"
    >
      <input
        onChange={[Function]}
        placeholder="Sol date"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    <div
      className="DayPickerInput"
    >
      <input
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        placeholder="YYYY-MM-DD"
        value=""
      />
    </div>
  </div>
  <div
    className="menu"
  >
    <div
      aria-expanded={false}
      className="ui item top left pointing dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      tabIndex={0}
    >
      <div
        aria-atomic={true}
        aria-live="polite"
        className="divider text"
        role="alert"
      >
        Camera
      </div>
      <i
        aria-hidden="true"
        className="dropdown icon"
        onClick={[Function]}
      />
      <div
        className="menu transition"
      >
        <div
          className="item"
          onClick={[Function]}
          role="option"
        >
          Default
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="option"
        >
          Front Hazard Avoidance Camera
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="option"
        >
          Rear Hazard Avoidance Camera
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="option"
        >
          Mast Camera
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="option"
        >
          Chemistry and Camera Complex
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="option"
        >
          Mars Hand Lens Imager
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="option"
        >
          Mars Descent Imager
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="option"
        >
          Navigation Camera
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="option"
        >
          Panoramic Camera
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="option"
        >
          Miniature Thermal Emission Spectrometer (Mini-TES)
        </div>
      </div>
    </div>
  </div>
  <div
    className="right item"
    onClick={[Function]}
  >
    <button
      className="ui primary button"
      onClick={[Function]}
    >
       Filtrar
    </button>
  </div>
</div>
`;

exports[`matches snapshot ListPhotos 1`] = `
Array [
  <div
    style={
      Object {
        "margin": "20px",
        "textAlign": "center",
      }
    }
  >
    <i
      aria-hidden="true"
      className="angle double left big link icon"
      onClick={[Function]}
    />
    <i
      aria-hidden="true"
      className="angle left big link icon"
      onClick={[Function]}
    />
    <div
      className="ui huge circular label"
      onClick={[Function]}
    />
    <i
      aria-hidden="true"
      className="angle right big link icon"
      onClick={[Function]}
    />
    <i
      aria-hidden="true"
      className="angle double right big link icon"
      onClick={[Function]}
    />
  </div>,
  <div
    style={
      Object {
        "margin": "auto",
        "maxWidth": "800px",
        "padding": "5px",
      }
    }
  >
    <div
      className="grid-container"
    />
  </div>,
]
`;

exports[`matches snapshot Pagination 1`] = `
<div
  style={
    Object {
      "margin": "20px",
      "textAlign": "center",
    }
  }
>
  <i
    aria-hidden="true"
    className="angle double left big link icon"
    onClick={[Function]}
  />
  <i
    aria-hidden="true"
    className="angle left big link icon"
    onClick={[Function]}
  />
  <div
    className="ui huge circular label"
    onClick={[Function]}
  />
  <i
    aria-hidden="true"
    className="angle right big link icon"
    onClick={[Function]}
  />
  <i
    aria-hidden="true"
    className="angle double right big link icon"
    onClick={[Function]}
  />
</div>
`;

exports[`matches snapshot RoverCard 1`] = `
<a
  className="ui card rover-card rover-selected"
  onClick={[Function]}
>
  <div
    className="ui image"
  >
    <img />
  </div>
  <div
    className="content"
  >
    <div
      className="center aligned header"
    >
      
    </div>
  </div>
</a>
`;

exports[`matches snapshot Rovers 1`] = `
<div
  className="ui centered cards"
>
  <a
    className="ui card rover-card"
    onClick={[Function]}
  >
    <div
      className="ui image"
    >
      <img
        src="https://mars.nasa.gov/system/feature_items/images/6037_msl_banner.jpg"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="center aligned header"
      >
        Curiosity
      </div>
    </div>
  </a>
  <a
    className="ui card rover-card"
    onClick={[Function]}
  >
    <div
      className="ui image"
    >
      <img
        src="https://s1.eestatic.com/2019/02/13/actualidad/Actualidad_375974696_114650383_1024x576.jpg"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="center aligned header"
      >
        Opportunity
      </div>
    </div>
  </a>
  <a
    className="ui card rover-card"
    onClick={[Function]}
  >
    <div
      className="ui image"
    >
      <img
        src="https://jplopez.net/wp-content/uploads/2019/09/rover-spirit.jpg"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="center aligned header"
      >
        Spirit
      </div>
    </div>
  </a>
</div>
`;

exports[`matches snapshot SinglePhoto 1`] = `
<div
  className="img_wrap_grid"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <img
    alt=""
    className="img-post"
    height="100%"
    width="100%"
  />
</div>
`;
